cpp/src/Slice/SliceUtil.cpp

```diff
@@ -921,3 +944,24 @@ optional<string> Slice::findMetadata(const string& directive, const map<string,
     auto match = metadata.find(directive);
     return (match != metadata.end() ? make_optional(match->second) : nullopt);
 }
+
+// Adds the metadata from m2 into m1. m2 is not altered by this operation, but m1 is.
+// If metadata is repeated, the values in m1 take precedence.
```

Comment:
Each metadata is a string, not a map entry, so what does precedence mean here?

For example, if one metadata is cs:type(Foo) and the other metadata is cs:type(Bar), both would be added. I think in this case you want to add only one, but this requires more structure to the metadata.

Reply:
Even if it's a list and there can be repeats, often we will still only use the first entry we find. For instance, if you write
`[cs:type(Foo)] [cs:type(Bar)] ...`
Obviously a member can't have both types at the same time. In this case `Foo` _has precedence_ here, meaning that is the actual type that will be used. It has precedence because it will be first in the list.
This is what my comment is alluding to.

But you're also right, that we would need extra structure to do a proper merge like this... I forgot that my metadata changes never got merged in to master. I'll change these for now until I do get to merge them in.

Reply:
I fixed the implementation of these so even repeated metadata is copied over, but I left the helper functions, since even though they're short, we use their logic in many places, and they're much less verbose than working with the lists directly.